### ISSUE TITLE
exit OSH with 'Ctrl-D' tries to render one last time, which can have side effects

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#430]
Found this from https://github.com/oilshell/oil/issues/427 - when you exit OSH with 'Ctrl-D', the prompt reader tries to render one last time.

Current behavior:
```
osh$ PS1='$(time sleep 1)'
real    1.013
user    0.006
sys     0.000
^D
real    1.010
user    0.000
sys     0.005
```

Requested behavior (same as bash):
```
osh$ PS1='$(time sleep 1)'
real    0m1.005s
user    0m0.002s
sys     0m0.003s
^D
```
Strangely, this only happens for Ctrl-D, using `exit` behaves as expected.

**Comment by jyn514** [comment]
Tracked this down to a hacky looking double while loop in core/main_loop.py. Going to take a stab at it.

**Comment by andychu** [comment]
Released with 0.7.pre1: https://www.oilshell.org/release/0.7.pre1/
